### DISCUSSION TITLE
Source Video Frame rate & output file quality.

### DISCUSSION THREAD

**Discussion by gateway** [rkhamilton/vqgan-clip-generator#13] [General]
I noticed in the code your that we define the source frame rate, which depending on how smart people are they might not know ifts 24, 30, 60 etc.. I know with ffmpeg we can get the fps with some python code.

```import ffmpeg

probe = ffmpeg.probe(filename_video)
video_info = next(s for s in probe['streams'] if s['codec_type'] == 'video')
fps = int(video_info['r_frame_rate'].split('/')[0])```

wouldn't this be a better solution and also remove the need to define source fps?

Also in the past I have always exported out to png for better image quality, granted larger file sizes but thoughts on using png or use the -qscale:v to control jpg output quality? 

--source https://stackoverflow.com/questions/10225403/how-can-i-extract-a-good-quality-jpeg-image-from-a-video-file-with-ffmpeg

**Comment by rkhamilton** [comment]
I’ll refer you to the docstring for the extraction method, and the [readme](https://github.com/rkhamilton/vqgan-clip-generator#vqgan_clipgenerate-function-arguments) that describes the parameters passed to generate methods.
*When extracting video frames from an existing video, this sets how many frames per second will be extracted. Interpolation will be used if the video's native framerate differs.*

The reason it’s done this way is that you can get some FFMPEG-interpolation-based video smoothing and slow motion effects. Say you have a 30fps video. Extract 10 fps from it. Restyle it. Then encode with input fps=10, output=30 and it will interpolate two-thirds of the frames. Or you can get a slow-motion conversion by:
Extract 30 fps.
Restyle
Re-encode with assumed input fps = 15, output 60fps. It will interpolate every 15 frames into one second of output with 60 frames, even though you extracted 30 fps, thereby giving you a half-speed output.

When it’s not desirable to play speed-changing games, you may still want to generate a highly stylized long video where you don’t want to restyle 30fps of the original. It gives you the option to iterate faster by restyling fewer frames and then use ffmpeg to interpolate between them. It’s just a different look to the output.

Edit: I’ll take a look at exporting video to png. That’s probably my preference I just didn’t think to set it up that way. In the mean time you can always plug in an os.system(‘ffmpeg -args’) call in your script instead of using my canned method. The generate.restyle methods take an input that’s just a list of file paths. PNGs should work fine. 

Edit 2: Given that we’re going to completely mangle these images by running them through VQGAN+CLIP it seems like it doesn’t add a lot of value to have a big emphasis on a high quality image data pipeline. It’s a different philosophy to me than if we were editing video.
